core/primitives-core/src/account.rs

```diff
@@ -94,6 +97,9 @@ pub struct AccountV2 {
     code_hash: CryptoHash,
     /// Storage used by the given account, includes account id, this struct, access keys and other data.
     storage_usage: StorageUsage,
+    /// Global contracts fields
+    global_contract_hash: Option<CryptoHash>,
```

Comment:
for `AccountV2` we want to have enum representation for contract code as discussed